pkg/controller/clusterpool/collections.go

```diff
@@ -395,6 +417,13 @@ func (cds *cdCollection) Stale() []*hivev1.ClusterDeployment {
 	return append(cds.unknownPoolVersion, cds.mismatchedPoolVersion...)
 }
 
+// RegisterNewCluster adds a freshly-created cluster to the cdCollection, assuming it is installing.
+func (cds *cdCollection) RegisterNewCluster(cd *hivev1.ClusterDeployment) {
+	cds.byCDName[cd.Name] = cd
+	cds.installing = append(cds.installing, cd)
+	cds.sortInstalling()
```

Comment:
is it better to just insert the cd to the right location instead of appending and then sorting. i'm assuming he installing was already sorted right?

Reply:
It was, yes.

The right place would be the front of the list, since we're sorting these by reverse age.

However, I had to figure that out by looking at the `sortInstalling` function. I'm preferring to sacrifice a little bit of performance for the benefit of keeping the ordering logic in a single spot. Like, if we decided to change the ordering later, we could easily change `sortInstalling` and forget to change this function, leading to some tough-to-find bugs.

(I suppose I could get rid of `sortInstalling` and instead make an `addInstalling` that's used from both here and [L299](https://github.com/openshift/hive/pull/1528/files#diff-bbf4d3cb24b8cac87951b84ca973747902c00d4a9d662648c8e760a37569581dR299). But then I'm writing my own insertion sort, which is likely to be way less efficient than the builtin.)